hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1363,11 +1363,17 @@ func validateAutoscaling(nodePool *hyperv1.NodePool) error {
 	return nil
 }
 
-func defaultNodePoolAMI(region string, releaseImage *releaseinfo.ReleaseImage) (string, error) {
-	// TODO: The architecture should be specified from the API
-	arch, foundArch := releaseImage.StreamMetadata.Architectures["x86_64"]
+func defaultNodePoolAMI(region string, specifiedArch string, releaseImage *releaseinfo.ReleaseImage) (string, error) {
+	switch specifiedArch {
+	case "amd64":
+		specifiedArch = "x86_64"
```

Comment:
@enxebre What do you mean by "all these values"? Are you asking only for `specifiedArch`, or are you wanting the `case` values to be constants as well? It could get messy having both:
`caseArm64`, `rhcosAarch64`, `caseAmd64` and `rhcosX86_64`

Reply:
I meant all, so no string literal is ever duplicated as x64_64 is atm. Feel free to solve that problem as you prefer, I would go with:
```
const (
	// ArchitectureAMD64 indicates AMD64 (x86_64).
	ArchitectureAMD64 = "amd64"
	// ArchitectureS390X indicates s390x (IBM System Z).
	ArchitectureS390X = "s390x"
	// ArchitecturePPC64LE indicates ppc64 little endian (Power PC)
	ArchitecturePPC64LE = "ppc64le"
	// ArchitectureARM64 indicates arm (aarch64) systems
	ArchitectureARM64 = "arm64"
)

var (
	// ArchAliases contains the RHCOS release metadata aliases for the different architectures supported as API input.
	ArchAliases = map[string]string{
		architectureAMD64: "x86_64",
		architectureARM64: "aarch64",
	}
)
```

Then in this function you don't need the switch case but only `ArchAliases[arch]`.